### PR TITLE
Updates to the buffered configured api

### PR DESCRIPTION
**Pull request by fwh-dc**
Fixes issues with the buffered configured api:
* Handle empty buffers as a no-op
* Rewrite of CertificateChainBuffer because the old implementation was not consistent with CertificateChainFile

New features:
* Adds a ca certificate bundle configurer
* Adds a certificate verification mode configurer

### CODE REVIEW

File: src/oatpp-openssl/configurer/CertificateBuffer.cpp [hunk 1]
```diff
@@ -26,8 +26,15 @@
 
 namespace oatpp { namespace openssl { namespace configurer {
 
+CertificateBuffer::CertificateBuffer(const std::string& certificateBuffer)
```
Comment (lganzzzo) [inline src/oatpp-openssl/configurer/CertificateBuffer.cpp]: Please change to `oatpp::String`:

```cpp
CertificateBuffer::CertificateBuffer(const oatpp::String& certificateBuffer)
```

**Review by lganzzzo — APPROVED** [review]
Looks great!

Please fix minor comments (please change method params type from `std::string` to `oatpp::String`) - just for the sake of API consistency.

And it's ready to merge

**Comment by fwh-dc** [comment]
I found some more copy paste errors and additional error logging and updated to use Oatpp string. Please have a look if you like it :)

**Review by lganzzzo — APPROVED** [review]

**Comment by lganzzzo** [comment]
Looks great!
Merged!

**Pull request merged**